src/securejoin.rs

```diff
@@ -139,14 +139,22 @@ async fn get_self_fingerprint(context: &Context) -> Result<Fingerprint> {
     Ok(key.dc_fingerprint())
 }
 
+// hack around the changed JSON accidentally used by an iroh upgrade, see #6518 for more details and for code snippet.
+// this hack is mainly needed to give ppl timme to upgrade and can be removed after some months (added 2025-02)
+pub(crate) fn fix_add_second_device_qr(qr: &str) -> String {
+    qr.replacen(",\"info\":{\"relay_url\":", ",\"relay_url\":", 1)
+        .replacen("\"]}}", "\"]}", 1)
+}
+
 /// Take a scanned QR-code and do the setup-contact/join-group/invite handshake.
 ///
 /// This is the start of the process for the joiner.  See the module and ffi documentation
 /// for more details.
 ///
 /// The function returns immediately and the handshake will run in background.
-pub async fn join_securejoin(context: &Context, qr: &str) -> Result<ChatId> {
-    securejoin(context, qr).await.map_err(|err| {
+pub async fn join_securejoin(context: &Context, qr_err: &str) -> Result<ChatId> {
+    let qr_fixed = fix_add_second_device_qr(qr_err);
+    securejoin(context, &qr_fixed).await.map_err(|err| {
```

Comment:
The change here is not necessary:

`securejoin()` passes `qr` to `check_qr` (and not to any other function), and you changed `check_qr` already to call `fix_add_second_device_qr()`.

It also seems a bit dangerous to unconditionally call `fix_add_second_device_qr()` here since will replace all `"]}}` with `"]}`, and I don't know whether there might be any other QR code which has `"]}}` in it, which would be broken like this.

So, if I didn't overlook anything in my reasoning, this change here should be reverted.

Reply:
yeah, you're right,  i mixed that up, that should've gone to add second device